src/configs/gemm-config.c

```diff
@@ -967,6 +967,22 @@ static void init_f32_gemm_config(void) {
       f32_gemm_config.mr = 4;
       f32_gemm_config.nr = 4;
     }
+  #elif XNN_ARCH_RISCV && XNN_ENABLE_RISCV_VECTOR
+    const struct xnn_hardware_config* hardware_config = xnn_init_hardware_config();
+    assert(hardware_config != NULL);
+    if (hardware_config->use_riscv_vector) {
+      f32_gemm_config.minmax.gemm[XNN_MR_TO_INDEX(7)] = xnn_init_hmp_gemm_ukernel((xnn_gemm_ukernel_fn) xnn_f32_gemm_minmax_ukernel_7x4v__rvv);
+      f32_gemm_config.minmax.gemm[XNN_MR_TO_INDEX(1)] = xnn_init_hmp_gemm_ukernel((xnn_gemm_ukernel_fn) xnn_f32_gemm_minmax_ukernel_1x4v__rvv);
+      f32_gemm_config.minmax.igemm[XNN_MR_TO_INDEX(7)] = xnn_init_hmp_igemm_ukernel((xnn_igemm_ukernel_fn) xnn_f32_igemm_minmax_ukernel_7x4v__rvv);
+      f32_gemm_config.minmax.igemm[XNN_MR_TO_INDEX(1)] = xnn_init_hmp_igemm_ukernel((xnn_igemm_ukernel_fn) xnn_f32_igemm_minmax_ukernel_1x4v__rvv);
+      f32_gemm_config.init.f32 = xnn_init_f32_minmax_scalar_params;
+      f32_gemm_config.pack_gemm_gio = (xnn_packw_gemm_gio_ukernel_fn) xnn_pack_f32_gemm_gio_w;
+      f32_gemm_config.pack_gemm_goi = (xnn_packw_gemm_goi_ukernel_fn) xnn_x32_packw_gemm_goi_ukernel_x4v__rvv_u8;
+      f32_gemm_config.mr = 7;
+      // nr is set to vlmax_e32m4 = 4*VLEN/SEW = 4*VLEN/32 = 4*VLENB*8/32 = VLENB
```

Comment:
So really its 4 vectors, so (vlen * 4)  (because m4), but expressed in elements, so vlen * 4 / sizeof(float).
You comment is hard to understand for people that dont know RVV terminology, so maybe use the vlen * 4 / sizeof(float) so it will be more clear when we do fp16 and other types or m2/m8 values.
7x4v makes sense to me.  7 works well on cpus with 8 way l1 cache, and 4 works well on cpus with 2 or 4 vector units.  There is likely no advantage to m8 over m4 on any hardware near term, and a drawback of using more registers, causing spill.
But if you can, test this PR after doing this config, using end2end_bench and/or use gemm_e2e_bench to confirm this is the best kernel.

Reply:
I've modified the comment.
Also, I'll follow your guide to do some tests on different m. I.e m(4~7) x 4v using end2end_bench and/or use gemm_e2e_bench.

Reply:
Do you think I can add different m size microkernels to XNNPACK like what you do for arm kernels?
I see you add 2x16__aarch64 ~ 6x16__arrch64.